src/main/java/com/bookstore/bookstoreapi/data/service/WishlistService.java

```diff
@@ -37,4 +40,29 @@ public Optional<Wishlist> createWishlist(String wishlistName, Long userID) {
 
     return Optional.of(newWishlist);
   }
+
+  public Optional<Wishlist> addBookToWishlist(Long bookID, Long wishlistID) {
+
+    Optional<Book> bookOptional = bookRepository.findById(bookID);
+    Optional<Wishlist> wishlistOptional = wishlistRepository.findById(wishlistID);
+
+    if (bookOptional.isEmpty() || wishlistOptional.isEmpty()) {
+      return Optional.empty();
+    }
+
+    Book book = bookOptional.get();
+    Wishlist wishlist = wishlistOptional.get();
+
+    if (wishlist.getBooks().contains(book)) {
+      return Optional.empty();
```

Comment:
I have to do it this way because the professor asks for an empty response, if I could return a proper response, it would be different. but for now, this IMO is sufficient.